vanilla-extract/app/root.tsx

```diff
@@ -8,16 +11,11 @@ import {
   ScrollRestoration,
 } from "@remix-run/react";
 
-import vanillaExtractStyles from "~/styles/index.css";
-
-export function links() {
+export const links: LinksFunction = () => {
   return [
-    {
-      rel: "stylesheet",
-      href: vanillaExtractStyles,
-    },
+    ...(cssBundleHref ? [{ rel: "stylesheet", href: cssBundleHref }] : []),
   ];
```

Comment:
I don't think we need spreading & we can just return the (empty) array?

```suggestion
  return cssBundleHref ? [{ rel: "stylesheet", href: cssBundleHref }] : [];
```